jcl/src/java.base/share/classes/java/lang/invoke/FilterArgumentsWithCombinerHandle.java

```diff
@@ -0,0 +1,91 @@
+    protected final MethodHandle combiner;
+    private final int filterPosition;
+    private final int[] argumentIndices;
+
+    FilterArgumentsWithCombinerHandle(MethodHandle handle, int filterPosition, MethodHandle combiner, int... argumentIndices) {
+        super(handle.type, KIND_FILTERARGUMENTS_WITHCOMBINER, infoAffectingThunks(combiner.type(), filterPosition, argumentIndices));
+        this.handle = handle;
+        this.combiner = combiner;
+        this.filterPosition = filterPosition;
+        this.argumentIndices = argumentIndices;
+    }
+
+    FilterArgumentsWithCombinerHandle(FilterArgumentsWithCombinerHandle originalHandle, MethodType newType) {
+        super(originalHandle, newType);
+        this.handle = originalHandle.handle;
+        this.combiner = originalHandle.combiner;
+        this.filterPosition = originalHandle.filterPosition;
+        this.argumentIndices = originalHandle.argumentIndices;
+    }
+
+    static FilterArgumentsWithCombinerHandle get(MethodHandle handle, int filterPosition, MethodHandle combiner, int... argumentIndices) {
+        return new FilterArgumentsWithCombinerHandle(handle, filterPosition, combiner, argumentIndices);
+    }
+
+    @Override
+    MethodHandle cloneWithNewType(MethodType newType) {
+        return new FilterArgumentsWithCombinerHandle(this, newType);
+    }
+
+    private static Object[] infoAffectingThunks(MethodType combinerType, int filterPosition, int...argumentIndices) {
+        Object[] result = {combinerType, filterPosition, argumentIndices};
+        return result;
+    }
+
+    private static final ThunkTable _thunkTable = new ThunkTable();
+    @Override
+    protected ThunkTable thunkTable(){ return _thunkTable; }
+    
+    @Override
+    protected final ThunkTuple computeThunks(Object info) {
+        return thunkTable().get(new ThunkKeyWithObjectArray(ThunkKey.computeThunkableType(type()), (Object[])info));
+    }
+
+    private static native int filterPosition();
+    private static native int argumentIndices();
+    /* create placeholder for combiner based on argumentIndices the original handle's arguments */
+    private static native int argumentsForCombiner(int indice, int argPlaceholder);
+    /* number of arguments remaining after the filtered argument */
+    private static native int numSuffixArgs();
+
+    @FrameIteratorSkip
+    private final int invokeExact_thunkArchetype_X(int argPlaceholder) {
+        if (ILGenMacros.isShareableThunk()) {
+            undoCustomizationLogic(combiner, handle);
+        }
+        if (!ILGenMacros.isCustomThunk()) {
+            doCustomizationLogic();
+        }
+
+        return ILGenMacros.invokeExact_X(handle, ILGenMacros.placeholder(
+            ILGenMacros.firstN(filterPosition(), argPlaceholder), 
+            ILGenMacros.invokeExact(combiner, argumentsForCombiner(argumentIndices(), argPlaceholder)),
+            ILGenMacros.lastN(numSuffixArgs(), argPlaceholder)));
+    }
+}
```

Comment:
This class has to implement `compareWith`. https://github.com/eclipse/openj9/blob/master/jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java#L1421

I think it should be similar or exactly the same as `FoldHandle`'s implementation